bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/BookieNettyServer.java

```diff
@@ -279,6 +280,12 @@ void start() {
     void shutdown() {
         LOG.info("Shutting down BookieNettyServer");
         isRunning.set(false);
+
+        if (!isClosed.compareAndSet(false, true)) {
```

Comment:
I did it in this way intentionly. Because current shutdown will be called twice, I would like to have logging to tell, so if we want to debug, we know what happened.